### ISSUE TITLE
Redis error ` Cannot assign requested address` when used with workspace container.

### ISSUE THREAD

**Issue by oberoivarun** [laradock/laradock#1378]
### Info:
- Docker version (`$ docker --version`): `17.12.0-ce, build c97c6d6`
- Laradock commit (`$ git rev-parse HEAD`): 0077de28a13feea669f9d46ef21c20477fa6dbb0
- System info (Mac, PC, Linux): Windows 10 Pro
- System info disto/version: No Idea

### Issue:

After `docker-compose up -d nginx mysql redis` All containers start without errors, but when i send a request to laravel, like an http ajax, or if i start `tinker`, or even the `queue:work` command, i get this error.

`PHP Warning:  stream_socket_client(): unable to connect to tcp://localhost:6379 (Cannot assign requested address) in /var/www/vendor/predis/predis/src/Connection/StreamConnection.php on line 127
Predis\Connection\ConnectionException with message 'Cannot assign requested address [tcp://localhost:6379]'`

But, if i stop the `nginx` container and just do a `php artisan serve` it works fine. There is some IP problem. So i checked the IP of the redis container. It was `172.20.0.4` and set the `REDIS_HOST` env variable to that. Still same error.
_____

### Expected behavior:

It should'nt give the error.

### Reproduce:

1. git clone laradock.
2. Initiate an empty laravel project and set it up with laradock.
3. Use redis as Cache driver and try to use it

**Comment by Drummer01** [comment]
Having same problem with workspace container. Besides, I am running node.js server which is running beyound laradock, and it connects to redis just fine.
@oberoivarun I checked and assigned IP of redis container to `REDIS_HOST` variable, all works fine for me.
I am using redis as `BROADCAST_DRIVER`

**Comment by agseco** [comment]
I had exact the same problem and I have solved it similarly as @Drummer01 said. In case it helps, these are the steps I followed:

1. Get the list of containers with ```docker ps```
2. Get the Redis container details with ```docker inspect <CONTAINER_ID>```
3. Get the IP address from Redis container details JSON (in my case located in ```NetworkSettings/Networks/laradock_backend/IPAddress```)
4. Set the IP address in the Laravel ```.env``` file (in my case property ```DB_HOST```)
